oga/backend/users/tests.py

```diff
@@ -1,3 +1,64 @@
-from django.test import TestCase
+from django.test import TestCase, Client
+from django.contrib.auth import get_user_model
+from .models import Location, Profile, Question
+import json
+from django.contrib.auth.models import User
+from django.contrib.auth import login, authenticate, get_user
+from django.views import generic
 
-# Create your tests here.
+class OgaTestCase(TestCase):
+    csrftoken = 0
+    sessionid = 0
+    client = Client()
+
+    def setUp(self):
+        User = get_user_model()
+        user = User.objects.create_user(username='test', password='1234')
+        user.save()
+        location = Location.objects.create(name='home',
+                                           longitude=38.123,
+                                           latitude=127.39)
+        location.save()
+        question = Question.objects.create(author=user, content='rains?',
+                                           location_id=location)
+        question.save()
+        profile = Profile.objects.get(user=user)
+        profile.location = location
+
+
+        self.client.login(username='test', password='1234')
```

Comment:
Since setup() is called before each test case, wouldn't the test User be already logged in before executing test_signup and test_signin?

Reply:
You are right. I will create AuthTestClass that tests singin/up without login in setUp, and test the others with the current setUp.